packages/snap-client/src/Client/apis/Abstract.test.ts

```diff
@@ -0,0 +1,265 @@
+import 'whatwg-fetch';
+import { API, ApiConfiguration, ApiConfigurationParameters, HTTPHeaders } from './Abstract';
+
+describe('Abstract Api', () => {
+	it('has expected default values', () => {
+		const api = new API(new ApiConfiguration({}));
+
+		//doesnt need any config
+		expect(() => {
+			new API(new ApiConfiguration({}));
+		}).not.toThrow();
+
+		expect(api.cache).toEqual({
+			config: {
+				enabled: true,
+				ttl: 300000, // ms
+				maxSize: 200, // KB
+				purgeable: true,
+			},
+			memoryCache: {},
+		});
+
+		// @ts-ignore
+		expect(api.configuration).toBeDefined();
+
+		// @ts-ignore
+		expect(api.configuration.configuration.maxRetry).toBe(8);
+
+		// @ts-ignore
+		expect(api.retryDelay).toEqual(1000);
+
+		// @ts-ignore
+		expect(api.retryCount).toEqual(1);
+
+		expect(api.cache).toBeDefined();
+
+		// @ts-ignore
+		expect(api.request).toBeDefined();
+
+		// @ts-ignore
+		expect(api.createFetchParams).toBeDefined();
+
+		// @ts-ignore
+		expect(api.fetchApi).toBeDefined();
+	});
+
+	const customHeaders: HTTPHeaders = {
+		customheader: 'customkey',
+	};
+
+	let CustomCacheConfig = {
+		ttl: 2222,
+		enabled: false,
+		maxSize: 4, // KB
+		purgeable: false,
+	};
+
+	// set up
+	const fetchfn = jest
+		.spyOn(global.window, 'fetch')
+		.mockImplementation(() => Promise.resolve({ status: 200, json: () => Promise.resolve() } as unknown as Response));
```

Comment:
I wonder if we can somehow use `Partial` here and wherever else `as unknown as Response` is being used...